tmc-langs-framework/src/main/java/fi/helsinki/cs/tmc/langs/domain/MetaSyntax.java

```diff
@@ -0,0 +1,52 @@
+package fi.helsinki.cs.tmc.langs.domain;
+
+public class MetaSyntax {
+    
+    private static final String BEGIN_SOLUTION = "BEGIN[ \\t]+SOLUTION";
+    private static final String END_SOLUTION = "END[ \\t]+SOLUTION";
+    private static final String SOLUTION_FILE = "SOLUTION[ \\t]+FILE";
+    private static final String SPACES = "[ \\t]*";
+    private static final String STUB = "STUB:[ \\t]*";
+   
+    private String commentStartRegex;
+    private String commentEndRegex;
+    private String beginSolutionRegex;
+    private String endSolutionRegex;
+    private String solutionFileRegex;
+    private String stubRegex;
+    private String stubMarker;
+
+    public MetaSyntax(String commentStartSyntax, String commentEndSyntax) {
+        this.commentStartRegex   = SPACES + commentStartSyntax + SPACES;
+        this.commentEndRegex     = SPACES + commentEndSyntax   + SPACES;
+        this.beginSolutionRegex  = commentStartRegex  + BEGIN_SOLUTION + commentEndRegex;
+        this.endSolutionRegex    = commentStartRegex  + END_SOLUTION   + commentEndRegex;
+        this.solutionFileRegex   = commentStartRegex  + SOLUTION_FILE  + commentEndRegex;
+        this.stubRegex           = commentStartRegex  + STUB + "(.*)"  + commentEndRegex;
+        this.stubMarker          = commentStartSyntax + STUB; // Initial spaces left outside stubMarker on purpose
```

Comment:
A nit picky stylistic note:

While horizontal alignment is not forbidden, it is not problem free. See f.ex. this quote from the style guide:

> Alignment can aid readability, but it creates problems for future maintenance. Consider a future change that needs to touch just one line. This change may leave the formerly-pleasing formatting mangled, and that is allowed. More often it prompts the coder (perhaps you) to adjust whitespace on nearby lines as well, possibly triggering a cascading series of reformattings. That one-line change now has a "blast radius." This can at worst result in pointless busywork, but at best it still corrupts version history information, slows down reviewers and exacerbates merge conflicts.

Unless horizontal alignment **greatly** improves legibility I'd suggest not doing it.